### PR TITLE
Extension/winappdriver support

### PR DESCRIPTION
**Pull request by mreiche**
# Description

Complete refactoring of `WebDriverRequest` API for `WebDriverFactory`.
That allows to enable module specific features for each request type, also improves logging and introduces a cleaner `WebDriver` lifecycle.

## Changes

* You should prefer using `WebDriverRequest` for creating new WebDrivers, like:
```java
WebDriverRequest webDriverRequest= new DesktopWebDriverRequest();
WebDriverRequest webDriverRequest= new AppiumDriverRequest();
WebDriverRequest webDriverRequest= new WinAppDriverRequest();

WebDriver webDriver = WEB_DRIVER_MANAGER.getWebDriver(webDriverRequest);
```
* You can unwrap these drivers like
```java
WEB_DRIVER_MANAGER.unwrapWebDriver(webDriver, AppiumDriver.class).ifPresent(appiumDriver -> {});
```
* All `WebDriver` related request information like baseUrl, SeleniumUrl are moved to `WebDriverRequest`. 
* Using `WebDriverManager.getConfig()` is now deprecated and may be removed in the future.
* `WebDriverFactory` lifecycle improvements:
  -  `prepareWebDriverRequest()`: Creates an specific `WebDriverRequest` from given `WebDriverRequest`. E.a. `UnspecificWebDriverRequest` -> `DesktopWebDriverRequest`
  - `createWebDriver()`: Creates the new WebDriver
  - `setupNewWebDriverSession()`: Setups the new session (like window maximize or open base url)
  - The `AbstractWebDriverFactory` is gone
* Required for WinAppDriver support in https://github.com/telekom/testerra-appium-connector/pull/5

### CODE REVIEW

File: driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java [hunk 1]
```diff
@@ -39,7 +38,7 @@ public DesktopWebDriverUtils() {
 
     }
 
-    public NodeInfo getNodeInfo(URL seleniumUrl, String sessionId) {
+    public URL getNodeInfo(URL seleniumUrl, String sessionId) {
```
[the remark this answers is not in the document]
Comment (mreiche) [reply driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java:41]: Selenoid connector doesn't have any more information about that. The NodeInfo class only contains host and port, but a Selenoid host ist actually adressed by an URL. When we need more information, we should query them on runtime.

[the remark this answers is not in the document]
Comment (martingrossmann) [reply driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java:41]: I mean the line 50. This is a specific Selenoid/GGR REST call (https://aerokube.com/ggr/latest/#_getting_host_by_session_id) and shoult not part of Testerra, but of Selenoid Connector.
But we should discuss this not in this PR.

**Review by martingrossmann — APPROVED** [review]

File: driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mreiche) [reply driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java:41]: I could migrate this.

[the remark this answers is not in the document]
Comment (martingrossmann) [reply driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverUtils.java:41]: Yes, but this should also part of Testerra 1.

**Pull request merged**